internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go

```diff
@@ -302,7 +301,14 @@ func (c *Controller) runTask(n int, f runtime.TaskSetupFunc, seq runtime.Sequenc
 
 	logger := &log.Logger{}
 
-	if err := kmsg.SetupLogger(logger, fmt.Sprintf("[talos] task %d:", n), true); err != nil {
+	machinedLog, err := c.Runtime().Logging().ServiceLog("machined").Writer()
+	if err != nil {
+		return err
+	}
+
+	defer machinedLog.Close() //nolint: errcheck
+
+	if err := kmsg.SetupLogger(logger, fmt.Sprintf("[talos] task %d:", n), machinedLog); err != nil {
```

Comment:
new kmsg logger setup with runtime logging interface (supports anything we pick up for the log implementation)